### PR TITLE
Link design

### LINKED ISSUE #69: Design: Navigation links
The Navigation Link component needs to be styled according to the [Figma design](https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn/Design-System-Gemeente-Den-Haag?node-id=1485%3A1156).

## Acceptance criteria
**Variants implemented**
- [x] Text only
- [x] Icon left
- [x] Icon right

**States implemented**
- [x] Normal
- [x] Hover
- [x] Disabled
- [x] Focus

**Requirements**
- [x] Icon is placed as a `prop` not in the children
- [x] [Alignment](https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn/Design-System-Gemeente-Den-Haag?node-id=1485%3A1156) requirements are met
- [x] Classes use the BEM naming convention
- [x] Design tokens are identified > proposed > implemented
- [x] Only the `props` and possible values for these `props` which are necessary in our use-case are exposed. All other Material-UI and standard HTML props which are not deemed necessary in our use-case should not be allowed/forwarded to the underlying component.

### PR DESCRIPTION
**Pull request by ogjtech**
closes #69

### CODE REVIEW

File: src/components/Timeline/package.json [hunk 1]
```diff
@@ -25,8 +25,7 @@
         "@gemeente-denhaag/baseprops": "^0.1.7",
         "@gemeente-denhaag/icons": "^0.1.0",
         "@gemeente-denhaag/typography": "^0.1.7",
-        "@material-ui/core": "^4.11.0",
-        "clsx": "^1.1.1"
+        "@material-ui/core": "^4.11.0"
     },
```
Comment (DanielvanVliet) [inline src/components/Timeline/package.json:29]: if you are going to remove clsx in this PR, also remove any imports of clsx in the Timeline component

**Review by DanielvanVliet — CHANGES_REQUESTED** [review]
The disabled Link is still clickable with tab focus + enter.
Perhaps adding the disabled attribute will prevent this?

Can you also add some padding-block to the focused state of Link?
image
After added padding-block:
Knipsel

File: src/components/Link/src/index.tsx [hunk 2]
```diff
@@ -1,69 +1,72 @@
...
-    | 'subtitle1'
-    | 'subtitle2'
-    | 'body1'
-    | 'body2'
-    | 'caption'
-    | 'button'
-    | 'overline'
-    | 'srOnly'
-    | 'inherit';
+  focus?: boolean;
```
[the remark this answers is not in the document]
Comment (bddjong) [reply src/components/Link/src/index.tsx:32]: In addition to Daniel's comment, this focus property makes the link _look_ like it's focussed, but is it actually?

**Review by bddjong — CHANGES_REQUESTED** [review]

File: src/components/Timeline/package.json (hunk 1 above)
Comment (GewoonMaarten) [reply src/components/Timeline/package.json:29]: I first installed it only for Timeline, but then realized I wanted to install it for every component. See commit 
d925c9f

File: src/components/Link/src/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (GewoonMaarten) [reply src/components/Link/src/index.tsx:32]: > Do we really need to able to always focus on a Link?

Probably not, but I added it for the same reason we add a `.denhaag-link--focus` style.

File: src/components/Link/src/link.css [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  outline: var(--denhaag-link-focus-outline);
+}
+
+.denhaag-link--hover,
+.denhaag-link:hover {
+  cursor: pointer;
+  color: var(--denhaag-link-hover-color);
+}
+
+.denhaag-link--disabled {
```
[the remark this answers is not in the document]
Comment (GewoonMaarten) [reply src/components/Link/src/link.css:25]: This didn't work for me. Typescript doesn't allow me to add the `disabled` property to the `<a>` element.

**Comment by GewoonMaarten** [comment]
> The disabled Link is still clickable with tab focus + enter.
> Perhaps adding the disabled attribute will prevent this?

If I set `tabIndex` to `-1` when `disabled` this should not happen anymore.

> 
> Can you also add some padding-block to the focused state of Link?
> image
> After added padding-block:
> Knipsel

Yes, but I'll have to wait until [this comment](https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn?node-id=1485:1156#93004750) is resolve to get the correct value of this padding.

File: src/components/Link/src/Link.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,194 @@
+import { ArgsTable, Canvas, Description, Meta, Story } from "@storybook/addon-docs/blocks";
```
[the remark this answers is not in the document]
Comment (GewoonMaarten) [reply src/components/Link/src/Link.stories.mdx]: updated

File: src/components/Link/package.json [hunk 5]
```diff
@@ -23,7 +23,10 @@
     },
     "dependencies": {
         "@gemeente-denhaag/baseprops": "^0.1.7",
-        "@material-ui/core": "^4.11.0"
+        "@gemeente-denhaag/icons": "^0.1.0"
```
[the remark this answers is not in the document]
Comment (GewoonMaarten) [reply src/components/Link/package.json]: updated

File: src/components/Link/src/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (bddjong) [reply src/components/Link/src/index.tsx:32]: I doubt this property adds any value

[the remark this answers is not in the document]
Comment (GewoonMaarten) [reply src/components/Link/src/index.tsx:32]: Its for show casing right. That's reason for the style `denhaag-link--focus` style. See also: https://github.com/nl-design-system/utrecht/blob/9426602746f324498f6723766feb72ae30af7a27/components/link/bem.stories.mdx

**Comment by bddjong** [comment]
> Yes, but I'll have to wait until [this comment](https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn?node-id=1485:1156#93004750) is resolve to get the correct value of this padding.

I think we can go ahead and go for a best possible implementation until we get an update from the designers. It's just a small token value change after that :D

File: src/components/Link/src/index.tsx [hunk 6]
```diff
@@ -1,68 +1,73 @@
...
-  color?: 'initial' | 'inherit' | 'primary' | 'secondary' | 'textPrimary' | 'textSecondary' | 'error';
-
-  /**
-   * The component used for the root node.
-   * Either a string to use a HTML element or a component.
-   */
-  component?: React.ElementType;
+import './link.css';
 
+export interface LinkProps extends BaseProps, React.AnchorHTMLAttributes<never> {
```
Comment (bddjong) [inline src/components/Link/src/index.tsx:8]: Perhaps use `extends Omit<BaseProps, 'classes' | 'otherPropYouDontWant'>` to get rid of unused properties from the baseprops interface. Check out [this line of code](https://github.com/nl-design-system/denhaag/pull/319/files#diff-6f6da8a603f347833a13728b5a02d3ced3d52edfd7e36c8e72bddc96c50f2cb8R7)

File: src/components/Link/src/Link.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+import { Paragraph } from "@gemeente-denhaag/typography";
+const ReactParagraph = (
+  <Paragraph>
+    It is possible to put <Link href="#">a link</Link> in a Paragraph element.
+  </Paragraph>
+);
+// This 👆 transpiles into this 👇 in the DOM
+<p class="utrecht-paragraph utrecht-paragraph--distanced">
+  It is possible to put <a href="#" class="denhaag-link" tabindex="0">a link</a> in a Paragraph element.
+</p>
```
Comment (bddjong) [inline src/components/Link/src/Link.stories.mdx:44]: We can remove this and use an HTML only story similar to what @mjcarsjens implemented on the Alert design PR

File: src/components/Link/src/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (GewoonMaarten) [reply src/components/Link/src/index.tsx]: Removed the focus property

File: src/components/Link/src/Link.stories.mdx [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+            xmlns="http://www.w3.org/2000/svg"
+            class="MuiSvgIcon-root"
+            focusable="false"
+            aria-hidden="true"
+            shape-rendering="auto"
+          >
+            <path
+              d="M11.707 18.707a1 1 0 01-1.414 0l-6-6a1 1 0 010-1.414l6-6a1 1 0 111.414 1.414L7.414 11H19a1 1 0 110 2H7.414l4.293 4.293a1 1 0 010 1.414z"
+              fill="currentColor"
+            ></path>
```
Comment (bddjong) [inline src/components/Link/src/Link.stories.mdx:186]: Would it be possible to include all variants in the html only story. It'd be easier for people only copying our bem structure.

**Review by bddjong — APPROVED** [review]

**Review by bddjong — APPROVED** [review]

**Review by GewoonMaarten — APPROVED** [review]

**Pull request merged**